posthog/warehouse/api/external_data_source.py

```diff
@@ -185,9 +185,38 @@ class Meta:
             "schemas",
             "prefix",
         ]
-        extra_kwargs = {
-            "job_inputs": {"write_only": True},
+
+    """
+    This method is used to remove sensitive fields from the response.
+    IMPORTANT: This method should be updated when a new source type is added with sensitive fields.
+    """
+
+    def to_representation(self, instance):
+        representation = super().to_representation(instance)
+
+        sensitive_keys = {
+            "api_key",
+            "client_secret",
+            "hubspot_refresh_token",
+            "hubspot_secret_key",
+            "passphrase",
+            "password",
+            "private_key",
+            "private_key_id",
+            "salesforce_integration_id",
+            "secret_token",
+            "ssh_tunnel_auth_type_passphrase",
+            "ssh_tunnel_auth_type_password",
+            "ssh_tunnel_auth_type_private_key",
+            "stripe_secret_key",
+            "zendesk_api_key",
         }
+        job_inputs = representation.get("job_inputs", {})
+        if isinstance(job_inputs, dict):
+            for key in sensitive_keys:
+                job_inputs.pop(key, None)
+
+        return representation
```

Comment:
I think it's far too easy to forget to update this. I think we should have an opt-in system vs opt-out where we have to define which keys **do** get sent down to the client. Yes, it's a longer list of keys, but it will almost certainly save us from exposing secret keys over the wire in plaintext 

(arguably, I'd say don't send any keys down to the client to be 100% safe)

Reply:
I'll update the list to be a whitelist for the providers we support thus far, I think the best way is to have these stored separately, but that ship has sailed a little bit. But yeah, I think we'd for sure forget to update this.